@xen-orchestra/lite/src/composables/collection-filter.composable.ts

```diff
@@ -4,22 +4,22 @@ import { type LocationQueryValue, useRoute, useRouter } from "vue-router";
 
 interface Config {
   queryStringParam?: string;
+  initialFilters?: string[];
 }
 
-export default function useCollectionFilter(
-  config: Config = { queryStringParam: "filter" }
-) {
+export default function useCollectionFilter<T>(config: Config = {}) {
   const route = useRoute();
   const router = useRouter();
-  const filtersSet = ref(
-    config.queryStringParam
-      ? queryToSet(route.query[config.queryStringParam] as LocationQueryValue)
-      : new Set<string>()
-  );
+  const { queryStringParam, initialFilters = [] } = config;
+  const filtersSet = ref<Set<string>>(new Set(initialFilters));
   const filters = computed(() => Array.from(filtersSet.value.values()));
 
-  if (config.queryStringParam) {
-    const queryStringParam = config.queryStringParam;
+  if (queryStringParam !== undefined) {
+    if (route.query[queryStringParam] !== undefined) {
+      filtersSet.value = queryToSet(
+        route.query[queryStringParam] as LocationQueryValue
```

Comment:
The [doc](https://router.vuejs.org/api/#locationquery) says: `route.query['foo']` return `LocationQueryValue | LocationQueryValue[]`
Are we sure we will never encounter an array?

Reply:
This shouldn't happen here since it's handled by the composable (unless a user has fun modifying the URL itself)

Reply:
What do you think of doing something like that, in order to handle this case and avoid triggering error.
```ts
const queryString = route.query[queryStringParam]
if(queryString !== undefined && !Array.isArray(queryString)){
   filtersSet.value = queryToSet(queryString)
} 

```

Reply:
Or we can take the first one in the array in that case. I'm guessing it's the first one that appears in the URL, it would be safe and good enough if it only happens if the user manually enters the URL.